### PR TITLE
Slotmanager

### PR DESCRIPTION
**Pull request by topnessman**

### CODE REVIEW

File: src/checkers/inference/DefaultSlotManager.java [hunk 1]
```diff
@@ -37,22 +39,61 @@
...
+     * ConstantSlot is subclass of VariableSlot. VariableSlot needs to be down
+     * cast to a subclass such as CombVariableSlot in order to get the concrete
+     * slot type
+     *
+     * @key id of slot used to locate a VariableSlot
+     * @value VariableSlot with this id(concrete type may be subclasses of
+     *        VariableSlot)
+     */
     private final Map<Integer, VariableSlot> variables;
+    /**
```
Comment (wmdietl) [inline src/checkers/inference/DefaultSlotManager.java]: To improve readability of code, add an empty line before a javadoc comment.

File: src/checkers/inference/DefaultSlotManager.java [hunk 2]
```diff
@@ -37,22 +39,61 @@
...
-    //a map of variable id to variable for ALL variables (including subtypes of VariableSlots)
+    /**
+     * A map of Integer to VariableSlot for storing all the slots encountered by
+     * this slot manager. If a slot is subtype of another slot, the subtype is
+     * also stored. ConstantSlots are also stored in this map, since
+     * ConstantSlot is subclass of VariableSlot. VariableSlot needs to be down
+     * cast to a subclass such as CombVariableSlot in order to get the concrete
+     * slot type
+     *
+     * @key id of slot used to locate a VariableSlot
```
Comment (wmdietl) [inline src/checkers/inference/DefaultSlotManager.java]: `@key` and `@value` would be useful javadoc tags for documenting maps, but I don't think they exist.
Can you try running javadoc on the code and see if this works?
If we don't have a `gradle javadoc` target yet, could you add one?

Comment (topnessman) [inline src/checkers/inference/DefaultSlotManager.java]: Thank you for your advice, professor!

**Comment by topnessman** [comment]
I add javadoc task to the build.gradle. But the javadoc task fails due to tons of errors and warnings during the process. Do we need to remove those errors?

**Comment by wmdietl** [comment]
The errors look like the Checker Framework dependencies are not found.
https://docs.gradle.org/current/dsl/org.gradle.api.tasks.javadoc.Javadoc.html#org.gradle.api.tasks.javadoc.Javadoc:classpath
gives the default classpath, which should work, but obviously doesn't.
Does it make a difference whether you first build CFI?
Can you try setting the classpath explicitly?

**Pull request merged**